### PR TITLE
PF33: The Pathfinders UI.

### PR DESCRIPTION
**Pull request by kberg**

### CODE REVIEW

File: src/styles/pathfinders.less [hunk 1]
```diff
@@ -1,3 +1,287 @@
...
+		background-position: -2px;
+	}
+	.track-tag-venus {background-image: url(./assets/tags/venus.png);}
+	.track-tag-earth {background-image: url(./assets/tags/earth.png);}
+	.track-tag-mars {background-image: url(./assets/tags/mars.png);}
+	.track-tag-jovian {background-image: url(./assets/tags/jovian.png);}
+	.track-tag-moon {background-image: url(./assets/tags/moon.png);}
+
+	.track-background-venus {
+		background: linear-gradient(to right, transparent 20%,#303030 40%,#303030), url("assets/buttons-homepage/planet8.png");
```
Comment (bafolts) [inline src/styles/pathfinders.less:45]: The file planet8.png is nearly 200kb. This seems larger than it needs to be for an image of this size. Can a lower quality version of the image be used?

File: src/client/components/pathfinders/PlanetaryTrack.vue [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<template>
+    <tr>
+      <td>{{icon}}</td>
+      <td v-for="idx in range" :key="idx" :class="getClass(idx)">
+        <Rewards :type="type" v-if="idx <= (rewards.spaces.length + 1) && rewards.spaces[idx] !== undefined" :rewards="rewards.spaces[idx]" :gameOptions="gameOptions" />
```
Comment (bafolts) [inline src/client/components/pathfinders/PlanetaryTrack.vue:5]: If `idx === rewards.spaces.length` or `idx === rewards.space.length + 1` would it be `undefined`? Is this checking that `rewards.spaces[idx]` is `undefined` when the value of `idx` is out of bounds?

File: src/client/components/pathfinders/Reward.vue [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  props: {
+    reward: {
+      type: String as () => Reward,
+    },
+    gameOptions: {
+      type: Object as () => GameOptionsModel,
+    },
+  },
+  computed: {
+    getClass(): String {
```
Comment (bafolts) [inline src/client/components/pathfinders/Reward.vue:26]: The type of this should be `string`.

File: src/client/components/pathfinders/Reward.vue [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        return 'resource_icon resource_icon--titanium';
+      case 'tr':
+        return 'tag-tr tag-size-big';
+      case 'venus_scale':
+        return 'venus-scale';
+      default:
+        return '';
+      }
+    },
+    outerClass(): String {
```
Comment (bafolts) [inline src/client/components/pathfinders/Reward.vue:75]: The type of this should be `string`.

File: src/client/components/pathfinders/Rewards.vue [hunk 5]
```diff
@@ -0,0 +1,56 @@
+<template>
+  <div class="rewards_cont">
+    <span v-if="mostTags">!&nbsp;</span>
+    <Reward v-for="(reward, idx) in myReward" :reward="reward" :key="'r' + idx" :gameOptions="gameOptions"/>
```
Comment (bafolts) [inline src/client/components/pathfinders/Rewards.vue:4]: Why is this `'r' + idx` needed for the key?

**Review by bafolts — APPROVED** [review]
I have minor comments. I would recommend using a lower image quality for the new planet*.png images. I assume they could all use a lower quality.

File: src/client/components/pathfinders/Reward.vue (hunk 3 above)
Comment (kberg) [reply src/client/components/pathfinders/Reward.vue:26]: Thanks, fixed.

File: src/client/components/pathfinders/Reward.vue (hunk 4 above)
Comment (kberg) [reply src/client/components/pathfinders/Reward.vue:75]: Thanks, fixed.

File: src/client/components/pathfinders/Rewards.vue (hunk 5 above)
Comment (kberg) [reply src/client/components/pathfinders/Rewards.vue:4]: I don't know! Removed it, and things look fine.

File: src/client/components/pathfinders/PlanetaryTrack.vue (hunk 2 above)
Comment (kberg) [reply src/client/components/pathfinders/PlanetaryTrack.vue:5]: Must be some legacy code from when I was struggling with dealing with the indexes. Removed the +1, and things still render.

File: src/styles/pathfinders.less (hunk 1 above)
Comment (kberg) [reply src/styles/pathfinders.less:45]: ## Current sizes: 

```
 96 planet3.png   68 planet5.png  152 planet9.png
 92 planet4.png  192 planet8.png 
```

Current render quality:

image


## Reduce to 75% of size:

```
136 planet3.png   92 planet5.png   80 planet9.png 
100 planet4.png   92 planet8.png 
```

Huh? Some got larger and some got smaller.

But it renders fine.

image

## Reduce to 50% of size:

```
64 planet3.png  44 planet5.png  40 planet9.png  
44 planet4.png  44 planet8.png 
```
image

I played with the css, adding background-size, width and height, but that did not help:

image

@ssimeonoff do you have any tips here?

Comment (kberg) [reply src/styles/pathfinders.less:45]: For the moment I am going to leave this, but I agree this should be addressed.

**Pull request merged**